test/general/key.js

```diff
@@ -2193,4 +2193,90 @@ VYGdb3eNlV8CfoEC
+    let privateKey;
+    let total;
+    return openpgp.generateKey(opt).then(function(key) {
+      privateKey = key.key;
+      total = privateKey.subKeys.length;
+      return privateKey.generateSubkey({passphrase: '123', sign: true});
+    }).then(function(answer){
+      expect(answer).to.exist;
+      expect(privateKey.subKeys.length).to.be.equal(total+1);
+      expect(answer).to.be.equal(privateKey.subKeys[total]);
+      const subkeyOid = answer.subKey.params[0];
+      const pkOid = privateKey.primaryKey.params[0];
+      expect(subkeyOid.getName()).to.be.equal(pkOid.getName());
+      expect(answer.subKey.algorithm).to.be.equal('eddsa');
+      return answer.verify(privateKey.primaryKey);
+    }).then(function(answer){
+      expect(answer).to.be.equal(openpgp.enums.keyStatus.valid);
+    });
+  });
+
+  it('create and add a new ec subkey to a rsa key', function() {
+    const privateKey = openpgp.key.readArmored(priv_key_rsa).keys[0];
+    privateKey.subKeys = [];
+    const total = privateKey.subKeys.length;
+    return privateKey.generateSubkey({passphrase: 'hello world', curve: 'ed25519'}).then(function(answer){
+      expect(answer).to.exist;
+      expect(privateKey.subKeys.length).to.be.equal(total+1);
+      expect(answer).to.be.equal(privateKey.subKeys[total]);
+      expect(answer.subKey.params[0].getName()).to.be.equal(openpgp.enums.curve.curve25519);
+      expect(answer.subKey.algorithm).to.be.equal('ecdh');
+      return answer.verify(privateKey.primaryKey);
+    }).then(function(answer){
+      expect(answer).to.be.equal(openpgp.enums.keyStatus.valid);
+    });
+  });
+
+  it('create and add a new subkey with different password', function() {
+    const privateKey = openpgp.key.readArmored(priv_key_rsa).keys[0];
+    const total = privateKey.subKeys.length;
+    return privateKey.primaryKey.decrypt('hello world').then(function(){
+      return privateKey.generateSubkey({passphrase: 'otherpass'});
+    }).then(function(answer){
+      expect(answer).to.exist;
+      expect(privateKey.subKeys.length).to.be.equal(total+1);
+      const subkeyN = answer.subKey.params[0];
+      const pkN = privateKey.primaryKey.params[0];
+      // console.log(subkeyN.data[0], pkN.data[0], openpgp.util.nbits(subkeyN.data[0]), openpgp.util.nbits(pkN.data[0]))
+      expect(subkeyN.byteLength()).to.be.equal(pkN.byteLength());
+      expect(answer).to.be.equal(privateKey.subKeys[total]);
+      expect(answer.subKey.algorithm).to.be.equal('rsa_encrypt_sign');
+      expect(answer.subKey.isDecrypted).to.be.false;
+      return answer;
+    }).then(function(answer){
+      return answer.subKey.decrypt('otherpass').then(function(){
+        expect(answer.subKey.isDecrypted).to.be.true;
+        return answer;
+      })
+    }).then(function(answer){
+      return answer.verify(privateKey.primaryKey);
+    }).then(function(answer){
+      expect(answer).to.be.equal(openpgp.enums.keyStatus.valid);
+    });
+  });
+
```

Comment:
Would be good to have tests that verify that the new keys can encrypt/sign/decrypt/verify as expected